biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/DTOFramework.java

```diff
@@ -0,0 +1,68 @@
+package biz.aQute.gogo.commands.provider;
+
+import java.util.List;
+
+import org.apache.felix.service.command.Descriptor;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.dto.BundleDTO;
+import org.osgi.framework.dto.FrameworkDTO;
+import org.osgi.framework.dto.ServiceReferenceDTO;
+
+import biz.aQute.gogo.commands.dtoformatter.DTOFormatter;
+
+public class DTOFramework {
+
+	private BundleContext context;
+
+	@SuppressWarnings("deprecation")
+	public DTOFramework(BundleContext context, DTOFormatter formatter) {
+		this.context = context;
+
+		// TODO: handle formatters
+		dtos(formatter);
+	}
+
+	void dtos(DTOFormatter formatter) {
+		formatter.build(FrameworkDTO.class)
+			.inspect()
+			.fields("*")
+			.line()
+			.part();
+
+	}
+
+
+	@Descriptor(value = "deliveres the FrameworkDTO")
+	public FrameworkDTO frameworkDTO() {
+		final Bundle bundle = context.getBundle(0);
+		final FrameworkDTO frameworkDTO = bundle.adapt(FrameworkDTO.class);
+		return frameworkDTO;
+	}
+
+	@Descriptor(value = "deliveres the BundleDTOs")
```

Comment:
spelling error,  happens several times in this file.

Reply:
thx done